ambry-rest/src/main/java/com.github.ambry.rest/RestServer.java

```diff
@@ -170,9 +173,14 @@ public RestServer(VerifiableProperties verifiableProperties, ClusterMap clusterM
             restServerConfig.restServerResponseHandlerScalingUnitCount, metricRegistry);
     restResponseHandler = restResponseHandlerFactory.getRestResponseHandler();
 
+    HelixPropertyStoreConfig helixStoreConfig = new HelixPropertyStoreConfig(verifiableProperties);
+    Notifier notifier = null;
+    if (helixStoreConfig.zkClientConnectString != HelixPropertyStoreConfig.INVALID_ZK_CLIENT_CONNECT_STRING) {
+      notifier = new HelixNotifier(helixStoreConfig);
```

Comment:
This is a tricky part. Besides `HelixAccountService`, we provide `InMemoryUnknownAccountService` for those Ambry adopters who do not want to migrate to containerization. The setup using `InMemoryUnknownAccountService` does not require a `notifier`, so it can be null. And instantiating a `notifier` depends on if we supply the zk config, that is why the check is here.